include/Engine/Rendering/ShadowPass.h

```diff
@@ -0,0 +1,82 @@
+#ifndef ShadowPass_h_
+#define ShadowPass_h_
+
+#include "IRenderer.h"
+#include "FrameBuffer.h"
+#include "ShaderProgram.h"
+#include "../Core/EventBroker.h"
+#include "../Core/World.h"
+#include "ShadowPassState.h"
+#include "imgui/imgui.h"
+
+#define MAX_SPLITS 4
+
+enum NearFar { NEAR = 0, FAR = 1 };
+enum LRBT { LEFT = 0, RIGHT = 1, BOTTOM = 2, TOP = 3 };
+
+struct ShadowFrustum
+{
+	float NearClip;
+	float FarClip;
+	float FOV;
+	float AspectRatio;
+	glm::vec3 MiddlePoint;
+	float Radius;
+	std::array<float, 4> LRBT;
+	std::array<glm::vec3, 8> CornerPoint;
+};
+
+class ShadowPass
+{
+public:
+	ShadowPass(IRenderer* renderer);
+	ShadowPass(IRenderer * renderer, int ShadowResX, int ShadowResY);
+	~ShadowPass();
+
+	void InitializeFrameBuffers();
+	void InitializeShaderPrograms();
+	void ClearBuffer();
+	void Draw(RenderScene& scene);
+
+	GLuint DepthMap() const { return m_DepthMap; }
+	std::array<glm::mat4, MAX_SPLITS> LightP() const { return m_LightProjection; }
+	std::array<glm::mat4, MAX_SPLITS> LightV() const { return m_LightView; }
+	std::array<float, MAX_SPLITS> FarDistance() const { return{ m_shadowFrusta[0].FarClip, m_shadowFrusta[1].FarClip, m_shadowFrusta[2].FarClip, m_shadowFrusta[3].FarClip }; }
```

Comment:
Will this break if we change max splits? Should be a for loop if that is the case.